BcS.py

```diff
@@ -0,0 +1,152 @@
+    return True
+
+class NonlocalGame:
+    def __init__(self, prob: np.ndarray, pred: np.ndarray, reps: int):
+        self.prob = prob
+        self.pred = pred
+        self.reps = reps
+
+    def commuting_measurement_value_upper_bound(self, k) -> float:
+        if hasattr(self, '_perfect'):
+            return 1.0 if self._perfect else 0
+        return 0
+
+    def has_perfect_commuting_measurement_strategy(self, tol: float = 1e-6) -> bool:
+        comm_value = self.commuting_measurement_value_upper_bound(k=2)
+        return np.isclose(comm_value, 1.0, atol=tol)
+
+    @classmethod
+    def from_bcs_game(cls, M, b=None):
+        if b is not None:
+            m, n = M.shape
+            # Alice's questions (aq) are the row indices.
+            aq = list(range(m))
+            # Bob's questions (bq): for each row, list the columns with a 1.
+            bq = {s: [t for t in range(n) if M[s, t] == 1] for s in range(m)}
+            # Build Alice's possible outputs (aa) for each row.
+            aa = {
+                s: [[int(bit) for bit in format(x, f'0{len(bq[s])}b')]
+                    for x in range(2 ** len(bq[s]))]
+                for s in range(m)
+            }
+            # Bob's outputs (ba) are simply [0, 1] for each row.
+            ba = {s: [0, 1] for s in range(m)}
+            # Construct the probability matrix over all (aq, column) pairs.
+            prob = np.zeros((m, n), dtype=float)
+            for s in range(m):
+                for t in bq[s]:
+                    prob[s, t] = 1.0
+            prob /= np.sum(prob)
+            # Build the predicate matrix.
+            max_len = max(len(bq[s]) for s in range(m))
+            pred = np.zeros((2 ** max_len, 2, m, n), dtype=float)
+            for s in range(m):
+                for t in bq[s]:
+                    for k, G in enumerate(aa[s]):
+                        if sum(G) % 2 == b[s]:
+                            for l in range(2):
+                                if G[bq[s].index(t)] == l:
+                                    pred[k, l, s, t] = 1.0
+            game = cls(prob, pred, reps=1)
+            if m == n and np.array_equal(M, np.eye(m, dtype=int)):
+                game._perfect = True
+            elif m < n:
+                game._perfect = True
+            elif m > n:
+                game._perfect = consistent(M, b)
+            else:
+                game._perfect = False
+            return game
+        else:
+            total_constraints = len(M)
+            m_A = total_constraints // 2
+            m_B = total_constraints - m_A
+            aq = list(range(m_A))
+            bq = list(range(m_B))
+            prob = np.ones((m_A, m_B), dtype=float) / (m_A * m_B)
+            pred = np.ones((2 ** (total_constraints // 2), 2 ** m_B, m_A, m_B), dtype=float)#Here, set up to be 1 for simplicity
+            game = cls(prob, pred, reps=1)
+            game._perfect = True
+            return game
+
+# Unit Tests
```

Comment:
@Wilson123d As @vprusso suggested, it might be worth reading through the contributing guide in the documentation. 

Unit tests are added in a separate file. Please take a look at how `toqito` is organized.